jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/CookieParser.java

```diff
@@ -57,4 +57,27 @@ default void parseFields(List<String> rawFields)
     {
         void addCookie(String name, String value, int version, String domain, String path, String comment);
     }
+
+    class InvalidCookieException extends IllegalArgumentException
```

Comment:
Why is this a nested private class?
It's exposed via API already.
Can this be a top level class in jetty-http please?

Reply:
It's not private.  It is public in the CookieParser interface.   We scope many things under interfaces in the API without making them top level classes.